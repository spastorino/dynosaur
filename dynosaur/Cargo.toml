[package]
name = "dynosaur"
version = "0.2.0"
authors = [
    "Santiago Pastorino <spastorino@gmail.com>",
    "Tyler Mandry <tmandry@google.com>",
    "Niko Matsakis <rust@nikomatsakis.com>",
]
categories = ["asynchronous", "no-std"]
keywords = ["async", "trait", "impl"]
description = "Dynamic dispatch for return position impl traits and async in Rust"
documentation = "https://docs.rs/dynosaur"
license = "MIT OR Apache-2.0"
repository = "https://github.com/spastorino/dynosaur"
edition = "2021"
rust-version = "1.75"

[dependencies]
dynosaur_derive = { version = "0.2", path = "../dynosaur_derive" }

[dev-dependencies]
tokio = { workspace = true }
trait-variant = { workspace = true }
ui_test = "0.28"

[[example]]
name = "next"

[[example]]
name = "trait_variant"

[[test]]
name = "ui"
harness = false
