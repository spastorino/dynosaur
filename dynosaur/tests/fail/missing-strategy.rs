use dynosaur::dynosaur;

#[dynosaur(DynMyTrait = dyn)]
trait MyTrait {
    async fn foo(&self) -> i32;
}

#[dynosaur(DynMyTrait2 = dyn MyTrait2)]
trait MyTrait2 {
    async fn foo(&self) -> i32;
}

fn main() {}
