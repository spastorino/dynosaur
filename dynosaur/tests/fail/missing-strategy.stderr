error: expected `dyn(box)`
 --> tests/fail/missing-strategy.rs:3:25
  |
3 | #[dynosaur(DynMyTrait = dyn)]
  |                         ^^^

error: expected `dyn(box)`
 --> tests/fail/missing-strategy.rs:8:26
  |
8 | #[dynosaur(DynMyTrait2 = dyn MyTrait2)]
  |                          ^^^

error: aborting due to 2 previous errors

